src/test/scala/edu/ie3/mobsim/MobilitySimulatorSpec.scala

```diff
@@ -308,32 +308,109 @@ class MobilitySimulatorSpec extends UnitSpec with MobilitySimulatorTestData {
       }
     }
 
-    "get time until next event" in {
-      val mobilitySimulator = mobSim()
-
-      val getTimeUntilNextEvent =
-        PrivateMethod[Long](Symbol("getTimeUntilNextEvent"))
+    "get time until next departure" in {
+      val getTimeUntilNextDeparture =
+        PrivateMethod[Option[Long]](Symbol("getTimeUntilNextDeparture"))
 
       val ev2: ElectricVehicle =
         evChargingAtSimonaWithStation.copy(departureTime =
           givenSimulationStart.plusMinutes(45)
         )
 
       val cases = Table(
-        ("evs", "expectedNextEvent"),
-        (Seq(arrivingEv), 3600),
+        ("evs", "expectedNextDeparture"),
+        (Seq(departingEv), 3600),
         (Seq(ev2), 2700),
-        (Seq(arrivingEv, ev2), 2700)
+        (Seq(departingEv, ev2), 2700)
+      )
+
+      forAll(cases) { (evs, expectedNextDeparture) =>
+        val actualNextDepartureOption =
+          MobilitySimulator invokePrivate getTimeUntilNextDeparture(
+            evs,
+            givenSimulationStart
+          )
+
+        val actualNextDeparture = actualNextDepartureOption.getOrElse(
+          fail("getTimeUntilNextDeparture returned None")
+        )
+
+        actualNextDeparture shouldBe expectedNextDeparture
+      }
+    }
+
+    "get time until next arrival" in {
+      val getTimeUntilNextArrival =
+        PrivateMethod[Option[Long]](Symbol("getTimeUntilNextArrival"))
+
+      val ev2: ElectricVehicle =
+        evChargingAtSimonaWithStation.copy(
+          parkingTimeStart = givenSimulationStart.plusMinutes(15)
+        )
+
+      val cases = Table(
+        ("evs", "expectedNextArrival"),
+        (Seq(arrivingEv), 1800L),
+        (Seq(ev2), 900L),
+        (Seq(arrivingEv, ev2), 900L)
       )
 
-      forAll(cases) { (evs, expectedNextEvent) =>
-        val actualNextEvent =
-          mobilitySimulator invokePrivate getTimeUntilNextEvent(
+      forAll(cases) { (evs, expectedNextArrival) =>
+        val actualNextArrivalOption =
+          MobilitySimulator invokePrivate getTimeUntilNextArrival(
             evs,
             givenSimulationStart
           )
 
-        actualNextEvent shouldBe expectedNextEvent
+        val actualNextArrival = actualNextArrivalOption.getOrElse(
+          fail("getTimeUntilNextArrival returned None")
+        )
+
+        actualNextArrival shouldBe expectedNextArrival
+      }
+    }
+
+    "get time until next event" in {
+      val getTimeUntilNextArrival =
+        PrivateMethod[Option[Long]](Symbol("getTimeUntilNextArrival"))
+      val getTimeUntilNextDeparture =
+        PrivateMethod[Option[Long]](Symbol("getTimeUntilNextDeparture"))
+
+      val evDeparting: ElectricVehicle =
+        ev1.copy(departureTime = givenSimulationStart.plusMinutes(45))
+      val evArriving: ElectricVehicle =
+        ev1.copy(
+          departureTime = givenSimulationStart.plusHours(18),
```

Comment:
Where does the 18 came from? Is that, when the ev is fully charged?

Reply:
I guess this part is mine. Just wanted to have an ev which arrives later as the first departs. Would have been easier to understand if I not copied twice from ev1. Will change this. The departureTime itself is just a random value.

Reply:
Maybe as an additional comment. The Mobsim sample the values for arrival and parkingTime both will lead to the departureTime where the car will leave. The time inbetween can be used for charging but don't need to. If the car is charged fully before, it will still wait the planned parkingTime.

Reply:
Ok, then it's clearer for me!